config/helm/kustomization.yaml

```diff
@@ -0,0 +1,44 @@
+namespace: '{{ .Release.Namespace }}'
+namePrefix: '{{- .Release.Name | replace "." "-" | trunc 33 | trimSuffix "-" -}}-'
+
+commonAnnotations:
+  meta.helm.sh/release-name: '{{ .Release.Name }}'
+  meta.helm.sh/release-namespace: '{{ .Release.Namespace }}'
+
+commonLabels:
+  app.giantswarm.io/branch: '{{ .Values.project.branch }}'
+  app.giantswarm.io/commit: '{{ .Values.project.commit }}'
+  app.kubernetes.io/managed-by: '{{ .Release.Service }}'
+  app.kubernetes.io/version: '{{ .Chart.AppVersion }}'
+  helm.sh/chart: '{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" -}}'
+  app.kubernetes.io/name: '{{ .Chart.Name | trunc 63 | trimSuffix "-" }}'
+  app.kubernetes.io/instance: '{{ .Release.Name }}'
+
+images:
+  - name: controller
+    newName: "{{ .Values.registry.domain }}/{{ .Values.image.name }}"
+    newTag: "{{ .Values.image.tag }}"
```

Comment:
This is all hackery but it works. It makes it almost on par with our current charts:

- Release namespace is used
- We use almost the same names (truncated to arbitrary number 33  instead of 49 to give more space for kustomize prefixes an pod suffixes)
- We have the same set of labels
- My local version of helm binary shouted at me so I added  `*helm.sh*` labels and annotations
- Selectors are equal to `commonLabels`
- Image is as we always set.

I couldn't make `.Values.pod.user.id` work because that's int value and (unlike `a: '{ .b }'`) `a: { .b }` is not a valid yaml. We have it set to a fixed value [here](https://github.com/giantswarm/capi-migration/blob/bb6c6764771b76cd32b0e821deb0f9776d0aac64/config/giantswarm/manager_security_context_patch.yaml#L10) and I think it's fine.